src/nwp_consumer/internal/repositories/model_repositories/ecmwf_realtime.py

```diff
@@ -0,0 +1,321 @@
+
+        Args:
+            path: The path to the grib file.
+        """
+        try:
+            dss: list[xr.Dataset] = cfgrib.open_datasets(path.as_posix())
+        except Exception as e:
+            return Failure(OSError(
+                f"Error opening '{path}' as list of xarray Datasets: {e}",
+            ))
+        if len(dss) == 0:
+            return Failure(ValueError(
+                f"No datasets found in '{path}'. File may be corrupted. "
+                "A redownload of the file may be required.",
+            ))
+
+        processed_das: list[xr.DataArray] = []
+        for i, ds in enumerate(dss):
+            try:
+                da: xr.DataArray = (
+                    entities.Parameter.rename_else_drop_ds_vars(
+                        ds=ds,
+                        allowed_parameters=ECMWFRealTimeS3ModelRepository.model().expected_coordinates.variable,
+                    )
+                    .rename(name_dict={"time": "init_time"})
+                    .expand_dims(dim="init_time")
+                    .expand_dims(dim="step")
+                    .to_dataarray(name=ECMWFRealTimeS3ModelRepository.model().name)
+                )
+                da = (
+                    da.drop_vars(
+                        names=[
+                            c for c in ds.coords
+                            if c not in ["init_time", "step", "variable", "latitude", "longitude"]
+                        ],
+                        errors="ignore",
+                    )
+                    .transpose("init_time", "step", "variable", "latitude", "longitude")
+                    .sortby(variables=["step", "variable", "longitude"])
+                    .sortby(variables="latitude", ascending=False)
+                )
+            except Exception as e:
+                return Failure(ValueError(
+                    f"Error processing dataset {i} from '{path}' to DataArray: {e}",
+                ))
+            # Put each variable into its own DataArray:
+            # * Each raw file does not contain a full set of parameters
+            # * and so may not produce a contiguous subset of the expected coordinates.
+            processed_das.extend(
+                [
+                    da.where(cond=da["variable"] == v, drop=True)
+                    for v in da["variable"].values
+                ],
+            )
+
+        return Success(processed_das)
+
+    @staticmethod
+    def _wanted_file(filename: str, it: dt.datetime, max_step: int) -> bool:
+        """Determine if the file is wanted based on the init time.
+
+        See module docstring for the file naming convention.
+        Returns True if the filename describes data corresponding to the input
+        initialization time and model metadata.
+
+        Args:
+            filename: The name of the file.
+            it: The init time of the model run.
+            max_step: The maximum step in hours to consider.
```

Comment:
I wonder if `it` in general should change to `init_time`, not to many more characters and its slightly easier to read